[lint] Actually make flake8 do something

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30892 [lint] Actually make flake8 do something**

Fixes all outstanding lints and actually installs a properly configured
flake8

Differential Revision: [D18862825](https://our.internmc.facebook.com/intern/diff/D18862825)

suo committed:
[lint] Actually make flake8 do something

Fixes all outstanding lints and actually installs a properly configured
flake8

albanD commented on test/dist_autograd_test.py:
```diff
@@ -1168,6 +1168,7 @@ def test_backward_simple_python_udf(self):
                 loss = ret.sum()
                 local_grads = self._verify_backwards(exec_mode, [loss], context_id, local_grads, t1, t2)
 
+    @staticmethod
```
How were these called before?

suo replied:
they are passed as function objects to the rpc system, which I think didn't treat them as methods (coincidentally)

albanD replied:
Ho ok... nice catch then :)

albanD commented on torch/jit/__init__.py:
```diff
@@ -1741,7 +1741,7 @@ def __contains__(self, key):
         # it is not overriden, we call into the nn.Module __dir__ method
         def __dir__(self):
-            self_method = getattr(self, "__dir__")
+            self_method = self.__dir__
```
Was there a good reason to do this? Why was getattr used to begin with?

suo replied:
I don't think so? They are basically the same when used with a string literal, so I guess it was just reflex.

albanD replied:
Ok, I'm not sure why you would ever use `getattr` with constant strings. That's why I was asking.
But the fix looks better yes.

albanD reviewed (approved):
Looks good, small questions.
Not sure if the fbgemm is expected though.

suo committed:
Update on "[lint] Actually make flake8 do something"


Fixes all outstanding lints and actually installs a properly configured
flake8

suo committed:
Update on "[lint] Actually make flake8 do something"


Fixes all outstanding lints and actually installs a properly configured
flake8

suo committed:
Update on "[lint] Actually make flake8 do something"


Fixes all outstanding lints and actually installs a properly configured
flake8

Differential Revision: [D18862825](https://our.internmc.facebook.com/intern/diff/D18862825)

kostmo commented:
## CircleCI build failures summary
As of commit c595231:
* **1/1** broken upstream at merge base 26c5146 (see [grid view](https://dr.pytorch.org/master-timeline.html?count=10&sha1=26c51468c5572869743346edaf017e00e93271b2&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
    * You may want to rebase on the `viable/strict` branch (see its [recency history](https://dr.pytorch.org/master-viable-commits.html)):
        * If your commit is older than `viable/strict`:
          ```
          git fetch viable/strict
          git rebase viable/strict
          ```
        * If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
          ```
          git fetch viable/strict
          git rebase --onto viable/strict $(git merge-base origin/master HEAD)
          ```
* **0/1** failures introduced in this PR



## Detailed failure analysis (WIP)

Here are the [reasons each build failed](https://dr.pytorch.org/commit-details.html?sha1=c5952312eb50cbf26a4f4c7d65ec80004b0d3fd5):

### 1 failure recognized by patterns:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3832856) pytorch_linux_xenial_py3_clang5_asan_build
**Step:** Build
```
Dec 06 21:34:04 caused by: Connection refused (os error 111)
```


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).
